extensions/vi-mode/commands.lisp

```diff
@@ -639,6 +640,12 @@ Move the cursor to the first non-blank character of the line."
         (apply-visual-range #'swapcase-region)
         (swapcase-region start end))))
 
+(define-command vi-swapcase-and-forward-char () ()
+  (let*  ((start (copy-point (current-point)))
+          (end (character-offset (current-point) 1)))
+    (vi-swapcase start end (current-state)))
```

Comment:
If you don't use copy-point in conjunction with delete-point, it will cause a memory leak, so it is better to use with-point.

If it were me, I would write it like this
```common-lisp
(with-point ((start (current-point))
             (end (current-point)))
  (character-offset end 1)
  ...)
```

Reply:
Please help me modify it, i just learn that.